cloud-infrastructure/ai-infra-gpu/Benchmarks/gds-benchmarks/README.md

```diff
@@ -0,0 +1,81 @@
+# gds-benchmakrs
+```
+[root@machine gds_benchmarks]# tree -L 2
+.
+├── backups
+│   ├── csv
+├── logs
+│   ├── CPU_GPU
+│   ├── CPU_ONLY
+│   ├── GPU_BATCH
+│   ├── GPU_DIRECT
+│   └── GPU_DIRECT_ASYNC
+├── README.md
+└── scripts
+    ├── 2csv.sh
+    ├── cufile.log
+    ├── initialize.sh
+    ├── launch.sh
+    └── rename.sh
+```
+
+The **'backups'** dir contains the backup of the previous runs (logfiles) and older CSV files. This directory is kept just in case some log files are missing.  
+The **'logs'** directory is designated for storing the most recent log files. Its sub-directories are organized according to the **'mode'** employed during the runs.
+
+The **'scripts'** directory contains all the bash scripts used for initializing the GDS machine, running benchmarks, and transforming log files into CSV format.
+
+**'initialize.sh'**: This script prepares the machine for GDS usage by performing necessary configurations. It disables ACS, sets the correct LBA for the NVMes, and handles the creation, formatting, and mounting of a RAID 0 array, while also adjusting file system privileges.
+
+**'launch.sh'**: This script executes the benchmarks and is parameterized by 'mode' (with options 0, 1, 2, 5, 6 corresponding to GPU_DIRECT, CPU_ONLY, CPU_GPU, GPU_DIRECT_ASYNC, GPU_BATCH, GPU_DIRECT) and the number of threads. Additionally, a third parameter allows for the specification of block size (ranging from 4k to 16M).
+
+During operation, **'launch.sh'** will conduct multiple tests using a variety of block and file sizes, which are predefined (Block sizes include: 4k, 8k, 16k, 32k, 64k, 128k, 256k, 512k, 1M, 2M, 4M, 8M, 16M). Each test iteration is performed four times. The file size varies depending on the number of threads, and to maintain efficiency, file sizes are kept relatively small. The GDSIO benchmark scales the file size with the number of threads, which can lead to extensive data processing. For example, running eight threads with a 128G file size would result in a total of 1024 TB of data being written, reflecting the scale of operations handled by this setup.
+
+
+Each test will be logged under the corresponding directory. The naming of the log files has the following format: ``gdsio_s<block size>_m<mode number>_w<threads / workers>_r<repetition number>_d<timestamp>.log``  
+gdsio_s1M_m2_w1_r1_d1671836744.log  
+For example, this log refers to a test that used 1M as block size, Mode 2 (CPU_GPU), 1 thread, and its timestamp.  
+Each log also contains the full command that was issued to run:  
+```
+/usr/local/cuda/gds/tools/gdsio -D /mnt/nvme0/ -d 0 -w 1 -s 8G -x 2 -i 1M -I 1 -V >> /home/ubuntu/gds_benchmarks/scripts/../logs/CPU_GPU//gdsio_s1M_m2_w1_r1_d1671836744.log  
```

Comment:
This expects a predefined user. Please don't expect any predefined user and make it more flexible by not using fixed usernames.